### PR TITLE
[RFC] ESLint + Prettier

### PR DESCRIPTION
**Pull request by Simonwep**
From our discussion in #192

I normally only use eslint, but prettier seems (except of ternary indentation, I cannot find any documentation about that feature in eslint) to work very well with eslint. The rules used in `.eslintrc` are mostly taken from [here](https://github.com/Simonwep/eslint-config-simon/blob/master/lib/base.js). 

The `.prettierrc`-file only contains `"singleQuote": true` because they use double-quotes as default.

ESLint already reported a few errors (Unused variables / imports) which I manually fixed.

I could't run tests because AVA somehow fails to work with node 13 and I only get ` test/*.test.js does not exist.`. Let's see if TravisCI manages to properly run it.

I've also updated the `.travis.yml` file:
``` Diff
  - travis_retry npm test
+ - npm run prettify
+ - npm run lint 
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,3 +19,6 @@ install:
 
 script:
 - travis_retry npm test
+- travis_retry npm test
```
Comment (jantimon) [inline .travis.yml]: why test twice?

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        "prefer-numeric-literals": "error",
+        "prefer-template": "error",
+        "prefer-spread": "error",
+        "prefer-rest-params": "error",
+        "yoda": "error",
+        "callback-return": "error",
+        "handle-callback-err": "error",
+        "no-process-exit": "warn",
+        "no-buffer-constructor": "error",
+        "no-mixed-requires": "warn",
```
Comment (jantimon) [inline .eslintrc]: Please don't use `warn` :)

Decide if it should be `right` or `wrong`.

Telling a developer that it is maybe wrong is not that helpful 😉

File: .travis.yml (hunk 1 above)
Comment (Simonwep) [reply .travis.yml]: Whups, that's a typo - sorry, got pretty late yesterday 😬

File: .eslintrc (hunk 2 above)
Comment (Simonwep) [reply .eslintrc]: Yeah, you're right but in some cases it maybe should't raise an error immediately (although I can't give you a quick example and if so you'd either harden the rule or allow it haha).
I'll mark all as error, never thought about this in that way so far :)

File: .travis.yml (hunk 1 above)
Comment (jantimon) [reply .travis.yml]: 👍

File: .eslintrc (hunk 2 above)
Comment (jantimon) [reply .eslintrc]: Cool 👍 if it is not bad enough for an error then it is probably okay to allow it 😉

**Pull request merged**